### PR TITLE
[ios_platform_images] Add ios platform images

### PR DESCRIPTION
**Pull request by gaaclarke**
## Description

This adds the plugin that will be useful for add2app clients that want to share assets between Flutter and their host application, specifically images.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/ios_platform_images/android/src/main/kotlin/io/flutter/ios_platform_images/IosPlatformImagesPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      val channel = MethodChannel(registrar.messenger(), "plugins.flutter.io/ios_platform_images")
+      channel.setMethodCallHandler(IosPlatformImagesPlugin())
+    }
+  }
+
+  override fun onMethodCall(call: MethodCall, result: Result) {
+    if (call.method == "getPlatformVersion") {
+      result.success("Android ${android.os.Build.VERSION.RELEASE}")
+    } else {
+      result.error("AndroidNotSupported", "This plugin is for iOS only.", null)
```
Comment (collinjackson) [inline packages/ios_platform_images/android/src/main/kotlin/io/flutter/ios_platform_images/IosPlatformImagesPlugin.kt]: Are you sure you want to make this plugin iOS-specific (as opposed to leaving the door open to supporting Android in the future)? It seems like the API is simple enough at this point to support a lowest common denominator Android reimplementation, although I think it makes sense to focus on iOS at the outset.

Possibly relevant wisdom (though it doesn't apply as much to packages that are outside the Flutter SDK)
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-exposing-api-cliffs
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-exposing-api-oceans

Comment (gaaclarke) [reply packages/ios_platform_images/android/src/main/kotlin/io/flutter/ios_platform_images/IosPlatformImagesPlugin.kt]: I argued this point and came around to the other side.  The conventional wisdom doesn't stand in the add2app scenario, it was formed around full Flutter apps.

The point of the plugin is to support sharing assets in an existing iOS app with a subset of the UI that is written in Flutter.  That means if the iOS API supplies 3 different ways of loading images, it isn't sufficient to support 2 of them and ask users to edit their iOS app in order to integrate Flutter.

This plugin is pointless in a non-add2app scenario.  When users want cross-platform image loading they should move their assets into Flutterland.

**Comment by collinjackson** [comment]
We've been using Java for our Android first-party plugins, so you should probably stick to that instead of Kotlin if you want to check in an empty stub.

File: packages/ios_platform_images/android/src/main/kotlin/io/flutter/ios_platform_images/IosPlatformImagesPlugin.kt (hunk 1 above)
Comment (collinjackson) [reply packages/ios_platform_images/android/src/main/kotlin/io/flutter/ios_platform_images/IosPlatformImagesPlugin.kt]: It sounds like you've made up your mind on this. I think if I were maintaining this I would lean towards either a very focused plugin (covering only iOS, only one direction) or a plugin that covers both iOS and Android and tries to cover as many image sharing use cases as possible. Right now you're sort of halfway and it isn't clear what functionality should be provided.

But I don't feel strongly about it and if you want to land it like this it's fine by me. I'd like @amirh to provide a second review though.

File: packages/ios_platform_images/android/src/main/kotlin/io/flutter/ios_platform_images/IosPlatformImagesPlugin.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  companion object {
+    @JvmStatic
+    fun registerWith(registrar: Registrar) {
+      val channel = MethodChannel(registrar.messenger(), "plugins.flutter.io/ios_platform_images")
+      channel.setMethodCallHandler(IosPlatformImagesPlugin())
+    }
+  }
+
+  override fun onMethodCall(call: MethodCall, result: Result) {
+    if (call.method == "getPlatformVersion") {
```
[the remark this answers is not in the document]
Comment (gaaclarke) [reply packages/ios_platform_images/android/src/main/kotlin/io/flutter/ios_platform_images/IosPlatformImagesPlugin.kt]: done

File: packages/ios_platform_images/example/test/widget_test.dart [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+// gestures. You can also use WidgetTester to find child widgets in the widget
+// tree, read text, and verify that the values of widget properties are correct.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:ios_platform_images_example/main.dart';
+
+void main() {
+  testWidgets('Verify Platform version', (WidgetTester tester) async {
```
[the remark this answers is not in the document]
Comment (gaaclarke) [reply packages/ios_platform_images/example/test/widget_test.dart]: done

**Comment by gaaclarke** [comment]
> We've been using Java for our Android first-party plugins, so you should probably stick to that instead of Kotlin if you want to check in an empty stub.

You sure?  I didn't chose Kotlin.  That is the what the template for plugins choses right now.  We actively encourage new plugins to be written in Kotlin.

**Comment by gaaclarke** [comment]
FYI the light is green now.

File: packages/ios_platform_images/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# IOS Platform Images
+
+A Flutter plugin to share images between Flutter and iOS.
```
Comment (xster) [inline packages/ios_platform_images/README.md]: Talk about densities. Something like A Flutter plugin to load images with the right pixel density between Flutter and iOS, or some such

File: packages/ios_platform_images/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    home: Scaffold(
+      body: Center(
+        child: Image(image: IosPlatformImages.load("flutter")),
+      ),
+      //..
+    ),
+  );
+}
+```
+
```
Comment (xster) [inline packages/ios_platform_images/README.md]: describe what this does. i.e. calls UIImage, finds image by that name in the bundle with the desired density.

File: packages/ios_platform_images/example/android/.gitignore [hunk 6]
```diff
@@ -0,0 +1,7 @@
+gradle-wrapper.jar
```
Comment (xster) [inline packages/ios_platform_images/example/android/.gitignore]: (optional) @amirh, do we support plugins with no android implementation and/or example currently?

File: packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.h [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#import <Flutter/Flutter.h>
+
+@interface IosPlatformImagesPlugin : NSObject <FlutterPlugin>
```
Comment (xster) [inline packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.h]: docs

File: packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.m [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+  [channel setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
+    if ([@"loadImage" isEqualToString:call.method]) {
+      NSString* name = call.arguments;
+      UIImage* image = [UIImage imageNamed:name];
+      NSData* data = UIImagePNGRepresentation(image);
+      if (data) {
+        result([FlutterStandardTypedData typedDataWithBytes:data]);
+      } else {
+        result(nil);
```
Comment (xster) [inline packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.m]: can we provide users with a bit more details in terms of what went wrong?

**Comment by xster** [comment]
LGTM

**Comment by collinjackson** [comment]
> > We've been using Java for our Android first-party plugins, so you should probably stick to that instead of Kotlin if you want to check in an empty stub.
> 
> You sure? I didn't chose Kotlin. That is the what the template for plugins choses right now. We actively encourage new plugins to be written in Kotlin.

I discussed offline with @amirh and we prefer to keep everything in Java and Objective-C for flutter/plugins for now until we're ready to start porting to Kotlin.

File: packages/ios_platform_images/README.md (hunk 4 above)
Comment (gaaclarke) [reply packages/ios_platform_images/README.md]: done

File: packages/ios_platform_images/README.md (hunk 5 above)
Comment (gaaclarke) [reply packages/ios_platform_images/README.md]: done

File: packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.h (hunk 7 above)
Comment (gaaclarke) [reply packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.h]: done

File: packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.m (hunk 8 above)
Comment (gaaclarke) [reply packages/ios_platform_images/ios/Classes/IosPlatformImagesPlugin.m]: Returning `nil` is the same contract `[UIImage imageNamed:]` follows.

File: packages/ios_platform_images/pubspec.yaml [hunk 9]
```diff
@@ -0,0 +1,64 @@
+name: ios_platform_images
+description: A new flutter plugin project.
```
Comment (collinjackson) [inline packages/ios_platform_images/pubspec.yaml]: Update this

Comment (gaaclarke) [reply packages/ios_platform_images/pubspec.yaml]: done

**Pull request merged**